crates/re_renderer/src/renderer/outlines.rs

```diff
@@ -67,7 +67,38 @@ use smallvec::smallvec;
 ///
 /// Object index 0 is special: It is the default background of each outline channel, thus rendering with it
 /// is a form of "active no outline", effectively subtracting from the outline channel.
-pub type OutlineMaskPreference = Option<[u8; 2]>;
+#[derive(Clone, Copy, Default)]
+pub struct OutlineMaskPreference(pub Option<[u8; 2]>);
+
+impl OutlineMaskPreference {
+    pub const NONE: OutlineMaskPreference = OutlineMaskPreference(None);
+
+    #[inline]
+    pub fn max(self, other: OutlineMaskPreference) -> Self {
+        match self.0 {
+            Some(self_val) => match other.0 {
+                Some(other_val) => OutlineMaskPreference(Some(self_val.max(other_val))),
+                None => self,
+            },
+            None => other,
+        }
+    }
```

Comment:
```diff
diff --git a/crates/re_renderer/src/renderer/outlines.rs b/crates/re_renderer/src/renderer/outlines.rs
index ee62c2c0e..557434dad 100644
--- a/crates/re_renderer/src/renderer/outlines.rs
+++ b/crates/re_renderer/src/renderer/outlines.rs
@@ -67,23 +67,12 @@ use smallvec::smallvec;
 ///
 /// Object index 0 is special: It is the default background of each outline channel, thus rendering with it
 /// is a form of "active no outline", effectively subtracting from the outline channel.
-#[derive(Clone, Copy, Default)]
+#[derive(Clone, Copy, Default, PartialEq, Eq, PartialOrd, Ord)]
 pub struct OutlineMaskPreference(pub Option<[u8; 2]>);
 
 impl OutlineMaskPreference {
     pub const NONE: OutlineMaskPreference = OutlineMaskPreference(None);
 
-    #[inline]
-    pub fn max(self, other: OutlineMaskPreference) -> Self {
-        match self.0 {
-            Some(self_val) => match other.0 {
-                Some(other_val) => OutlineMaskPreference(Some(self_val.max(other_val))),
-                None => self,
-            },
-            None => other,
-        }
-    }
-
     #[inline]
     pub fn some(channel_a: u8, channel_b: u8) -> Self {
         OutlineMaskPreference(Some([channel_a, channel_b]))
diff --git a/crates/re_viewer/src/misc/selection_state.rs b/crates/re_viewer/src/misc/selection_state.rs
index f8bc6912e..c220d489d 100644
--- a/crates/re_viewer/src/misc/selection_state.rs
+++ b/crates/re_viewer/src/misc/selection_state.rs
@@ -444,7 +444,7 @@ impl SelectionState {
                         } else {
                             &mut outline_mask.overall
                         };
-                        *outline_mask_target = outline_mask_target
+                        *outline_mask_target = (*outline_mask_target)
                             .max(OutlineMaskPreference::some(next_selection_mask_index(), 0));
                     }
                 }
@@ -511,7 +511,7 @@ impl SelectionState {
                         } else {
                             &mut outlined_entity.overall
                         };
-                        *outline_mask_target = outline_mask_target
+                        *outline_mask_target = (*outline_mask_target)
                             .max(OutlineMaskPreference::some(0, next_selection_mask_index()));
                     }
                 }
```

Reply:
convincing! Didn't know this was possible